app/controllers/api/v1/users_controller.rb

```diff
@@ -120,4 +120,43 @@ def update
     standard_update(User, current_human_user.id)
   end
 
-end
\ No newline at end of file
+  ###############################################################
+  # find_or_create
+  ###############################################################
+
+  api :POST, '/find-or-create', 'Finds or Creates a pending user a user.'
+  description <<-EOS
+    Either finds or creates a new user a user. The new user will be created
+    with the state set to "unclaimed" meaning that it is a place-holder for
+    an user who has not yet completed the sign up process.
+
+    An email address must be supplied, and an optional username may be as well.
+
+    If the username or email is already in use by an unclaimed user,
+    a user will not be created and the existing the user's ID is returned
+    #{json_schema(Api::V1::UnclaimedUserRepresenter, include: :readable)}
+  EOS
+  param :email, String, required: false,
+        desc: "Email address to search by or assign to newly created user"
+  param :username, String, required: false,
+        desc: "Username to search by or assign to newly created user"
+  param :password, String, required: false,
```

Comment:
Are these actual URL parameters instead of posted JSON data?  I don't think we can do that because that means these fields will be sent in the clear outside of the encryption offered by SSL.  We of course can't send credentials in the clear like that.